WikiWatershed/model-my-watershed
# Simplify Stream Totaling for Staging Compatibility

## Overview

For reasons currently not understood, the namedtuple result of `sum_subbasin_stream_lengths` is interpreted as a dict by Celery when run locally, but as a list when run on staging.

By outputting a dict always, we ensure that it is interpreted uniformly across platforms.

I've currently manually applied this patch on staging to get it to work.

Connects #2811

### Demo

Staging woking with the patch:

![image](https://user-images.githubusercontent.com/1430060/39677877-625d8d40-5150-11e8-9f21-0d4f6337654e.png)

### Notes

I was unable to find out exactly why this was happening, but I did find some discussion on conflicting behavior of converting namedtuples to dicts here: https://stackoverflow.com/questions/26180528/python-named-tuple-to-dictionary. In the interest of expediency, I settled for a solution sans explanation. 🚀🐦

## Testing Instructions

* On `develop`, run subbasin HUC-10 locally and save the table data as a CSV
* Check out this branch and run `bundle`
* Clear your local Redis cache by running: `vagrant ssh services -c 'redis-cli -n 1 --raw KEYS ":1:geop_*" | xargs redis-cli -n 1 DEL'`
* Run subbasin HUC-10 localy and export the table as a CSV again
* Compare the two CSVs and ensure they are identical
